hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -500,6 +511,9 @@ protected String defaultWarehouseLocation(TableIdentifier tableIdentifier) {
         return String.format("%s/%s", databaseData.getLocationUri(), tableIdentifier.name());
       }
 
+    } catch (NoSuchObjectException e) {
+      throw new NoSuchNamespaceException(
+          e, "Namespace does not exist: %s", tableIdentifier.namespace().levels()[0]);
```

Comment:
Just to double check, the exceptions might be different with Hive2 and Hive3.
Do we run the tests with both Hive2 and Hive3 on CI?

Reply:
I think we run Hive-CI with hive2 and hive3 both. Also I have validated manually for both versions. 
Both [Hive2](https://github.com/apache/hive/blob/release-2.3.9-rc0/metastore/src/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java#L137) and [Hive3](https://github.com/apache/hive/blob/release-3.1.3-rc3/standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/HiveAlterHandler.java#L143) have the same error message.

Reply:
Thanks!